connectors/concur/src/test/java/com/vmware/connectors/concur/ConcurControllerTest.java

```diff
@@ -283,6 +284,37 @@ void testConcurRegex() throws Exception {
         testRegex("expense_report_id", fromFile("concur/regex-input.txt"), expectedList);
     }
 
+    @Test
+    public void testAuthSuccess() {
+        mockConcurServer.expect(requestTo("/oauth2/v0/token"))
+                .andExpect(method(POST))
+                .andExpect(MockRestRequestMatchers.content().contentTypeCompatibleWith(APPLICATION_FORM_URLENCODED))
+                .andRespond(withSuccess(oauthToken, APPLICATION_JSON));
+
+        webClient.head()
+                .uri("/test-auth")
+                .header(AUTHORIZATION, "Bearer " + accessToken())
+                .header("x-concur-authorization", "Basic dXNlcm5hbWU6cGFzc3dvcmQ=")
+                .exchange()
+                .expectStatus().isNoContent();
+    }
+
+    @Test
+    public void testAuthFailure() {
+        mockConcurServer.expect(requestTo("/oauth2/v0/token"))
+                .andExpect(method(POST))
+                .andExpect(MockRestRequestMatchers.content().contentTypeCompatibleWith(APPLICATION_FORM_URLENCODED))
+                .andRespond(withStatus(HttpStatus.FORBIDDEN));
```

Comment:
That's no good.  It will mess up our contract with the clients.  I've created a PR into your branch to fix this: https://github.com/vmware/connectors-workspace-one/pull/130

Reply:
Also note, this was an existing problem w/ the cards & actions calls.  The client would not realize it needs to display a re-auth card b/c they wouldn't have received 400 backend-status 401.

Reply:
After the fix, you can get reauth cards like expected (I had to disable the test-auth to get past client validation):

![screen shot 2018-08-10 at 9 23 12 am](https://user-images.githubusercontent.com/2766717/43960242-6f3c7466-9c7f-11e8-956c-303a3fe927e4.png)

Reply:
And with the test-auth endpoint, client is validating the user+pass:

![screen shot 2018-08-10 at 9 29 36 am](https://user-images.githubusercontent.com/2766717/43960408-ea875384-9c7f-11e8-9d9d-950eb3588deb.png)